### PR TITLE
Add EndRod

### PR DESCRIPTION
**Pull request by inxomnyaa**
## Introduction

It adds end rods as decoration and light source
Supports vanilla-like rotation and exact same placement

### Behavioural changes

new block  ¯\_(ツ)_/¯

## Tests

Tested and used since month in a fork. No issues known

### CODE REVIEW

File: src/pocketmine/block/EndRod.php [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+			5 => 4,
+		];
+		$this->meta = ($target->getId() === self::END_ROD && $faces[$face] == $target->getDamage()) ? Vector3::getOppositeSide($faces[$face]) : $faces[$face];
+		$this->getLevel()->setBlock($block, $this, true, true);
+		return true;
+	}
+
+	public function getDrops(Item $item): array{
+		return [
+			[$this->id, 0, 1],
```
Comment (jasonwynn10) [inline src/pocketmine/block/EndRod.php]: this needs changed according to 00afc5c363c8e54fd8f82dfdb3a3bff76672de9d

**Review by jasonw4331 — CHANGES_REQUESTED** [review]

**Review by dktapps — CHANGES_REQUESTED** [review]
This causes players to be kicked for flying when walking on top of end-rods. Also see @jasonwynn10 's comment about the drops.

File: src/pocketmine/block/EndRod.php (hunk 1 above)
Comment (thebigsmileXD) [inline src/pocketmine/block/EndRod.php]: i see

**Comment by thebigsmileXD** [comment]
@dktapps "This", what exactly, i can't see it for a strange reason

**Comment by TheDiamondYT1** [comment]
If i were you @dktapps i wouldnt accept block pull requests untill the new json system is in place

File: src/pocketmine/block/EndRod.php [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+ *
+*/
+
+namespace pocketmine\block;
+
+use pocketmine\item\Item;
+use pocketmine\math\Vector3;
+use pocketmine\Player;
+
+class EndRod extends Flowable{
```
Comment (jasonwynn10) [inline src/pocketmine/block/EndRod.php]: Shouldn't extend flowable

**Review by jasonw4331 — CHANGES_REQUESTED** [review]

**Comment by jasonwynn10** [comment]
@TheDiamondYT1 this is a PR on the api3/blocks branch, so it will be added with everything else in the rewrite.

**Comment by thebigsmileXD** [comment]
Bounding box fixed.

**Comment by thebigsmileXD** [comment]
Opening a new PR due to git/github issues

**Pull request closed without merging**